runtime/bcutil/ROMClassBuilder.cpp

```diff
@@ -774,10 +774,13 @@ ROMClassBuilder::prepareAndLaydown( BufferManager *bufferManager, ClassFileParse
 			 */
 			if (!sharedStoreClassTransaction.isCacheFull()) {
 				if ( sharedStoreClassTransaction.allocateSharedClass(&sizeRequirements) ){
-					U_8 *romClassBuffer = (U_8*)sharedStoreClassTransaction.getRomClass();
+					J9ROMClass *romClassBuffer = (J9ROMClass*)sharedStoreClassTransaction.getRomClass();
 					/*
 					 * Make note that the laydown is occurring in SharedClasses
 					 */
+
+					romClassBuffer->extraModifiers |= J9AccClassIsStoredInSCC;
```

Comment:
Isn't this overwritten by the `extraModifiers` passed into `finishPrepareAndLaydown()` below which does not have `J9AccClassIsStoredInSCC` set ?

Reply:
For classes with no class path information (such as anon classes), we do `compareROMClassForEquality()` to ensure they are the exact same so that the cached version can be safely returned. If the cached `romclass->extraModifiers` has `J9AccClassIsStoredInSCC` set and `extraModifiers` passed to `compareROMClassForEquality()` does not have it, the comparsion will always fail.